src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -249,6 +251,12 @@ def export_config(cmd,
     if skip_keyvault:
         src_kvs = [keyvalue for keyvalue in src_kvs if keyvalue.content_type != KeyVaultConstants.KEYVAULT_CONTENT_TYPE]
 
+    if destination == 'appservice' and export_as_reference:
+        if endpoint is None:
+            endpoint = get_store_endpoint_from_connection_string(connection_string) or resolve_store_metadata(cmd, name)[1]
```

Comment:
This is where I was thinking we can validate if an endpoint has been resolved or not:
```
endpoint = get_store_endpoint_from_connection_string(connection_string) or resolve_store_metadata(cmd, name)[1]
if endpoint:
     src_kvs = [__map_to_appservice_config_reference(kv, endpoint) for kv in src_kvs]
else:
    logger.warning("Cannot export as AppConfig references because store endpoint could not be resolved. Key-values will be exported as-is.")
```